### ISSUE TITLE
Installation makes ssh login bash not run init files anymore

### ISSUE THREAD

**Issue by rugk** [pipxproject/pipx#456]
**Describe the bug**
As soon as you follow the installation instructions

**How to reproduce**
Follow the official instructions for Linux:
```
python3 -m pip install --user pipx
python3 -m pipx ensurepath
```

Not required to reproduce, but you may then do
```
pipx completions
echo 'eval "$(register-python-argcomplete pipx)"' >> ~/.bashrc
```

**What happens**
It seems to not run `.bashrc` anymore, when I login via SSH.
When I start a terminal on my lxcd desktop, it works, however.

**Expected behavior**

Don't modify my shell without asking me…

**System**

pipx v0.15.4.0
Debian 10
Raspian

**Workaround**

You can manually type `bash` and it runs a subshell that works…
Still awkward…

**Questions**

How can I now get my main shell back and get bash to run `.bashrc`??

**Comment by itsayellow** [comment]
It's hard to debug this based on the information provided.  There's no way that I know of pipx would stop a `.bashrc` from executing.  

`python3 -m pipx ensurepath` uses userpath (https://github.com/ofek/userpath) to add a couple of lines at the end of one or more shell rc files (possibly `.bashrc`, `.bash_profile`, or `.profile`, etc.) that adds pipx and pipx apps to your `PATH` environment variable.

I would check your shell startup files (especially the ends of them) to make sure nothing is awry.  userpath adds a comment with any lines it adds to let you know that it was responsible, so it should be easy to see any userpath additions.  You could remove the userpath additional lines and see if your environment works the way you expect again.

If you find that userpath messed something up, please let us know!  (And let the folks at userpath know.)  It's possible there is a userpath bug.

**Comment by cs01** [comment]
I'm going to close this. I can't think of a way for pipx to mess with bash like this, so I think this is unrelated. Let us know if you find anything else out and we can re-open if needed.

**Comment by dmoralesvaldivia** [comment]
## TL;DR
#### If `ensurepath` creates `.bash_profile` this prevents `.profile` from running on login, which usually runs `.bashrc`


## Explanation
Hi @itsayellow and @cs01

I would like to suggest reopening this issue as I bumped into it as well, and I think I might have spotted the reason why `.bashrc` might not load anymore when bash is invoked as a login shell (i.e. ssh) after `python3 -m pipx ensurepath` is run.

As @itsayellow mentioned, when `python3 -m pipx ensurepath` command is run, `pipx` appends the following to the `.bashrc` and `.bash_profile` files in the user's home directory:

```
# Created by `pipx` on $DATE
export PATH="$PATH:/home/$USER/.local/bin"
```

However, if the `.bash_profile` file is not found, `pipx` will create it and append the content above to it too.

The catch here is that, when `bash` is invoked as a login shell (ssh), it doesn't look for `.bashrc`. It looks for `.bash_profile` first and, only when it is not found, looks for another `bash`-related file: `.profile`. This is important to note because of two reasons:

1. Many home directories implement only the `.bashrc` and `.profile` files by default.
2. The `.profile` file often runs `.bashrc` by default.

For example, this is the content of my vanilla `.profile` file:

```
# if running bash
if [ -n "$BASH_VERSION" ]; then
    # include .bashrc if it exists
    if [ -f "$HOME/.bashrc" ]; then
	. "$HOME/.bashrc"
    fi
fi

# set PATH so it includes user's private bin if it exists
if [ -d "$HOME/bin" ] ; then
    PATH="$HOME/bin:$PATH"
fi

# set PATH so it includes user's private bin if it exists
if [ -d "$HOME/.local/bin" ] ; then
    PATH="$HOME/.local/bin:$PATH"
fi
```

So, if someone wasn't using `.bash_profile` before, `pipx ensurepath` will prevent the running of `.profile`, and therefore probably `.bashrc` too, when it creates the `.bash_profile` file that doesn't do anything else other than update the path.

My fix was simply to delete `.bash_profile`. This worked because `.bashrc` also contains the path update command anyway.
 
Hope this is helpful!

**Comment by itsayellow** [comment]
That is helpful, thanks for supplying the mechanism.

Adding to the path by hand can be a confusing process for novice users, so the idea was to use the `userpath` python package to help automate it for those users.  As you point out there can be nuances with different configurations that create problems for an automated mechanism.

I wonder if changing the installation directions to add a blurb about "add the following directories to your PATH if you are an advanced user and know how to do this" might at least be a moderate documentation step to help people who don't need the `pipx ensurepath` functionality.  Unfortunately novice users are the same ones who might not know how to add to their PATH AND would have trouble with an automated system failing.

Maybe more work needs to happen on the userpath package.

**Comment by itsayellow** [comment]
Relevant: ofek/userpath#19

**Comment by dmoralesvaldivia** [comment]
Hi @itsayellow , thank you for your comments and for re-opening the issue.

As you mentioned and linked, I think you're right that the userpath package's behaviour is causing this issue, so I guess a solution would be to either change the package's behaviour or to not use it in the ensure_path function.

Either way, I also think your idea to add a note about this in the installation instructions is good :)

**Comment by itsayellow** [comment]
My PR in userpath: https://github.com/ofek/userpath/pull/36

**Comment by ofek** [comment]
https://pypi.org/project/userpath/1.5.0/

**Comment by itsayellow** [comment]
Thanks @ofek for the quick release!

**Comment by itsayellow** [comment]
I'm going to close this now that #684 is merged.  It should fix the problem described by the OP and others.  When the next version of pipx comes out (>0.16.2.1) let us know if there are still problems.
